### ISSUE TITLE
Surge not loading osx10.11 logic 9.1

### ISSUE THREAD

**Issue by 88pidhpidh** [surge-synthesizer/surge#3691]
**Bug Description:**
**After downloading the dmg, the plug-in does not appear in the DAW and cannot be found.**

**Surge Version**
**Surge Version 1.8.0. First download. No versions previously installed.** 

* Version:
* **Plugin Type: Installed all available. In my case VST3, AU, VST3 for standalone FX and AU for standalone.**
* Bitness:  **64- Bit**

**Reproduction Steps:**
**It didn't come to that**

**Expected Behavior:**
**Open the plug in normally like any other. If that is not possible or complicated steps are necessary, this plug-in is unattractive.**

**Screenshots:**
**Not possible because the plug-in cannot be found.**

**Computer Information (please complete the following information):**
 - OS: **OSX 10.11.6**
 - Host: **Logic Pro**
 - Version: **9.1.8**

**Additional Information:**

**Comment by mkruselj** [comment]
Did you perhaps try to restart the computer? It's sometimes a thing for AU plugins that has to be done, this started happening at a certain macOS version I don't remember which one...

**Comment by baconpaul** [comment]
Logic 918 is a very old version. Don’t know we have ever tested in it. Are you configured to load other 64 bit plugins?

**Comment by 88pidhpidh** [comment]
@mkruselj Hello, yes I have. Then I checked whether all the folders, like other plug-ins, were there. Unfortunately, not all folders were listed in the specific places. I think the installer may not include everything. As already described, I have version OS 10.11. El Capitan.

**Comment by baconpaul** [comment]
OK then it is almost definitely that our folder doesn’t force a rescan and cache rebuild and that’s required on older logics.

Here’s my advice

0: install surge from the installer then reboot your system once. If after a reboot it doesn’t work then:
1: you are running backups right? Make sure you do before you take any advice from internet folks
2. Quite logic and all other audio programs (audacity, AU lab, etc...)
3: There’s a logic AU cache you need to rebuild. You want to rebuild it. Safest way it to move it (so if rebuild fails you can move it back). To do that try in terminal

```
mv ~/Library/Caches/AudioUnitCache ~/Desktop
```

And restart logic. That should do a rescan of all your audio units and surge should appear with the rest of your units. If it does, remove that file from your desktop - it’s not needed any more. If something goes horribly wrong, quit logic again and move the file back. 

If it doesn’t appear then the next thing we have to do is figure out if the scan throws an error on a macos that old. There’s other steps for that which I can share but perhaps try this first.

**Comment by baconpaul** [comment]
(There is a small chance that surge doesn’t work with logic 9 on OS X 10.11 also; we have never tested it. But that’s my best guess)

**Comment by 88pidhpidh** [comment]
@baconpaul **It´s old but Gold.** Logic 9.1.8 läuft bei mir sehr gut und stabil. Ich kann mich nicht wirklich beschweren. Diese Version beinhaltet auch eine 32- Bridge. Ich kann damit beides verwenden, also 64 und 32 Bit. If I were to run on OSX 10.12 Sierra this DAW would not run as well as it does now. El Capitan is basically the last OSX with Logic 9.1.8. is still going very well.

**Comment by baconpaul** [comment]
Oh yeah sure i understand that. But it doesn’t mean that we have ever tested surge with it :)

**Comment by 88pidhpidh** [comment]
@baconpaul Thanks for your tips and hints. I will try to do it according to your description. Fortunately, I always have backups running in parallel. I will tell you how it went then. Hope it will be useful for others too.

**Comment by 88pidhpidh** [comment]
Hello Baconpaul,

I took all the steps you mentioned. I can't find the path ~ / Library / Caches / AudioUnitCache ~ / Desktop in this way. There is a folder with Surge under ~ / Library / Application Support.

Under ~ / Library / Audio / Plug-Ins there is also a folder with Surge, VST3 and Components.

There is also no surge folder in the chaches.

So much so far. Incidentally, I noticed with this installer that it takes a long time in contrast to other software instruments. Possibly a discrepancy. 
(Tried to show the paths here with a picture, but unfortunately cannot upload any pictures)

**Comment by 88pidhpidh** [comment]
...sorry that it took longer. I haven't had time yet.

**Comment by baconpaul** [comment]
What does this command show?

```
auval -vt aumu VmbA
```

**Comment by 88pidhpidh** [comment]
Command shows -> No such file or directory

**Comment by baconpaul** [comment]
Oh darn. So osx 10.11 doesn’t have au val. Hmm.

Lemme ponder. 

One other thing. Can you consider downloading hosting au (http://ju-x.com/hostingau.html) which is a tiny AU host I use for testing. See if it finds surge when you start it?

**Comment by 88pidhpidh** [comment]
The host has been loaded and it finds Surge.

**Comment by 88pidhpidh** [comment]
Ok, the Mini-Daw has found the plug-in, what is the solution? Because Logic 9.1.8 still doesn't find it. Did the DAW create a folder and I just have to move something?

**Comment by baconpaul** [comment]
OK that’s great. So surge works on osx11 just not yet in your logic. So hmm. I really am pretty sure that it is your audio unit cache. 

Can you quit logic and hostingau, start a terminal and type this

```
ls ~/Library/Caches/Audio*
```

And let me know what it shows? If you do have a file there called “AudioUnitCache” you need to rename it.

Logic X works that way, i remember Logic 9 working that way, and the internet lets me think that’s what the file is. But it MIGHT be a file there named ‘com.apple.audiounits.cache’ if your logic is old enough.

Basically “How do you force Logic Pro 9.1.8 to do a full rescan” is the problem you need to solve. If HAU loads it and LP918 doesn’t that’s the missing step.

**Comment by 88pidhpidh** [comment]
Unfortunately .. No such file or directory. I also looked for it in the folders and paths, without success

**Comment by 88pidhpidh** [comment]
Can you maybe tell me how to cleanly uninstall the plug-in and then try again?

**Comment by baconpaul** [comment]
Huh I wonder where Logic Pro 9 keeps stuff.

Anyway to uninstall there’s a couple of ways but the surest is, in terminal do 

```
sudo rm -rf /Library/Audio/Plug-Ins/Components/Surge.component
sudo rm -rf /Library/Application Support/Surge
```

Then grab the DMG and run the installer again

But that won’t solve your problem. The installer will just put those files back. You need to figure out how to force Logic to rescan., If Hosting AU runs it then either (1) LP9.1.8 won’t run surge ever for some reason or (2) you need to force a rescan somehow, which involves knowing how to do that in LP9.1.8

Does the 918 plugin manager have a ‘rescan all plugins’ option? I’ve been using LPX for nearly 8 years now so I doin’t remember how that worked.

**Comment by 88pidhpidh** [comment]
Logic 9 has the function to re-scan plug-ins via the AU Manager. But it is also possible that previously known ones are only scanned. Therefore, I will first delete Surge, then delete the audiocache completely. Then reinstall Surge. This is the only way Logic will perform a full scan. It will take a long time, but trying is more than studying. I report what came out in the end.

**Comment by 88pidhpidh** [comment]
hello paul, unfortunately did not help. audiocache cleared, then complete rescan of logic 9.1.8 and the plug in is not found in the DAW.

**Comment by baconpaul** [comment]
Right
OK so then I actually don't know what to suggest.
The last thing I would say is: blow away the audio cache, then run logic from the terminal so you can see output and see if it gives any message about surge.

To run logic from the terminal you, in a terminal, do

```
/Applications/Logic Pro/Contents/MacOS/Logic
```

or some such (I don' know the exact names for Logic 9.1.8)

But it also might be that: Surge doesn't work with Logic that old, and there's no way for me to fix it, unfortunately.

**Comment by mkruselj** [comment]
I bet JUCE will resolve this one, too.

**Comment by mkruselj** [comment]
@88pidhpidh We are closing this issue due to inactivity. If you can confirm that Surge XT now works for you, great! If you don't, again great. 🙂
